src/scoped.component.js

```diff
@@ -68,111 +66,85 @@ export class Scoped extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.css) {
-      if (this.state.css !== this.props.css) {
-        this.doneWithCss();
-        this.setState(this.newCssState(this.props))
-      }
-    } else {
-      if (prevProps.postcss.id !== this.props.postcss.id
-        || prevProps.postcss.styles !== this.props.postcss.styles
-        || prevProps.postcss.namespace !== this.props.postcss.namespace) {
-        this.doneWithPostcss();
-        this.setState(this.newPostcssState(this.props));
-      }
+    const oldCss = prevProps.postcss || prevProps.css
+    const newCss = this.props.postcss || this.props.css
+    if (
+      oldCss !== newCss ||
+      oldCss.id !== newCss.id ||
```

Comment:
for strings, it's still okay to check `'str'.id !== 'str'.id`. So this check can work with both css and postcss